rajawali/src/main/java/c/org/rajawali3d/core/AFrameDelegate.java

```diff
@@ -0,0 +1,234 @@
+                if (callback.callFrameStart()) {
+                    frameStartCallbacks.add(callback);
+                }
+                if (callback.callFrameEnd()) {
+                    frameEndCallbacks.add(callback);
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean removeFrameCallback(@NonNull final FrameCallback callback) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                if (callback.callFrameStart()) {
+                    frameStartCallbacks.remove(callback);
+                }
+                if (callback.callFrameEnd()) {
+                    frameEndCallbacks.remove(callback);
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean clearFrameCallbacks() {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                frameStartCallbacks.clear();
+                frameEndCallbacks.clear();
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean addAnimation(final Animation anim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.add(anim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean removeAnimation(final Animation anim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.remove(anim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean replaceAnimation(final Animation oldAnim, final Animation newAnim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.set(animations.indexOf(oldAnim), newAnim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean addAnimations(final Collection<Animation> anims) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.addAll(anims);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean clearAnimations() {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.clear();
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    //
+    // ACoreComponent method overrides
+    //
+
+    @RenderThread
+    @Override
+    @CallSuper
+    public void onFrameStart(double deltaTime) throws InterruptedException {
+        if (!isEnabled) {
```

Comment:
In this case I am with @rpicolet, I routinely add code prior to calling a super method for the very specific reason of manipulating the data the super method will be acting on, and in my interpretation of OOP, this is very much an intended consequence of inheritance.